homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -0,0 +1,188 @@
+"""Support for Sure PetCare Flaps/Pets binary sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_LOCK,
+    DEVICE_CLASS_PRESENCE,
+    BinarySensorDevice,
+)
+from homeassistant.const import CONF_ID, CONF_NAME, CONF_TYPE
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    CONF_DATA,
+    DATA_SURE_PETCARE,
+    DEFAULT_DEVICE_CLASS,
+    SPC,
+    TOPIC_UPDATE,
+    SureLocationID,
+    SureLockStateID,
+    SureThingID,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up Sure PetCare Flaps sensors based on a config entry."""
+    entities = []
+
+    spc = hass.data[DATA_SURE_PETCARE][SPC]
+
+    for thing in spc.ids:
+        sure_id = thing[CONF_ID]
+        sure_type = thing[CONF_TYPE]
+        sure_data = thing[CONF_DATA]
```

Comment:
Remove this line. The data will be available under `spc.states`. We will pass the `spc` instance to the entity.